Sources/Widgets/Widgets3D/LineWidget/behavior.js

```diff
@@ -26,70 +37,79 @@ export default function widgetBehavior(publicAPI, model) {
+  );
+
   function ignoreKey(e) {
     return e.altKey || e.controlKey || e.shiftKey;
   }
 
+  function updateCursor() {
+    model.widgetState.setIsDragging(true);
+    model.openGLRenderWindow.setCursor('grabbing');
+    model.interactor.requestAnimation(publicAPI);
+  }
+
+  // --------------------------------------------------------------------------
+  // Text Methods
+  // --------------------------------------------------------------------------
+
   /*
+   * DetectOffsetDirectionForTextPosition
    * check for handle 2 position in comparison to handle 1 position
    * and sets text offset to not overlap on the line representation
    */
-  function calcTextPosWithLineAngle() {
+
+  function detectOffsetDirectionForTextPosition() {
     const pos1 = model.widgetState.getHandle1().getOrigin();
     const pos2 = model.widgetState.getHandle2().getOrigin();
-    const SVGTextProps = model.representations[2].getTextProps();
 
     let dySign = 1;
     if (pos1[0] <= pos2[0]) {
       dySign = pos1[1] <= pos2[1] ? 1 : -1;
     } else {
       dySign = pos1[1] <= pos2[1] ? -1 : 1;
     }
-    SVGTextProps.dy = dySign * Math.abs(SVGTextProps.dy);
+    return dySign;
   }
 
-  function updateHandleDirection(behavior, callData) {
-    let bv = behavior;
-    if (bv === HandleBehavior.HANDLE1_ALONE) {
-      const handle1Pos = model.widgetState.getHandle1().getOrigin();
-      const WorldMousePos = publicAPI.computeWorldToDisplay(
-        model.renderer,
-        handle1Pos[0],
-        handle1Pos[1],
-        handle1Pos[2]
-      );
-      const mousePos = publicAPI.computeDisplayToWorld(
-        model.renderer,
-        callData.position.x,
-        callData.position.y,
-        WorldMousePos[2]
-      );
-      vtkMath.subtract(
-        model.widgetState.getHandle1().getOrigin(),
-        mousePos,
-        Direction
-      );
-      bv = 0;
-    } else {
-      const modifier = bv === 1 ? 1 : -1;
-      bv -= 1;
-      const handle1Pos = model.widgetState.getHandle1().getOrigin();
-      const handle2Pos = model.widgetState.getHandle2().getOrigin();
-      vtkMath.subtract(handle1Pos, handle2Pos, Direction);
-      vtkMath.multiplyScalar(Direction, modifier);
-    }
-    model.representations[bv].getGlyph().setDirection(Direction);
+  function calcTextPosWithLineAngle() {
+    const dySign = detectOffsetDirectionForTextPosition();
+    const textPropsCp = { ...model.representations[2].getTextProps() };
+    textPropsCp.dy = dySign * Math.abs(textPropsCp.dy);
+    model.representations[2].setTextProps(textPropsCp);
   }
 
+  publicAPI.setText = (text) => {
+    if (model.widgetState.getText()) {
+      model.widgetState.getText().setText(text);
+    }
+    model.interactor.render();
+  };
+
+  publicAPI.updateTextPosition = (positionOnLine) => {
+    model.widgetState.setPositionOnLine(positionOnLine);
+    updateTextPosition(model, positionOnLine);
+    model.interactor.render();
+  };
+
+  publicAPI.setPositionOnLine = macro.chain(
```

Comment:
who/where calls `setPositionOnLine` ?

Maybe you should instead observe when `widgetState.PositionOnLine` is modified

Reply:
this function is called in the UI

Reply:
Problem is, some public functions are declared in index.js and some in behavior.js.
It is not very developer friendly.